dbos/queue.py

```diff
@@ -0,0 +1,40 @@
+import threading
+import time
+from typing import TYPE_CHECKING, Optional
+
+from dbos.core import P, R, _execute_workflow_id, _start_workflow
+from dbos.error import DBOSInitializationError
+
+if TYPE_CHECKING:
+    from dbos.dbos import DBOS, Workflow, WorkflowHandle
+
+
+class Queue:
+    def __init__(self, name: str, concurrency: Optional[int] = None) -> None:
+        self.name = name
+        self.concurrency = concurrency
+        from dbos.dbos import _get_or_create_dbos_registry
+
+        registry = _get_or_create_dbos_registry()
+        if self.name in registry.queue_info_map:
+            raise DBOSInitializationError(
+                f"Failed to register queue {self.name}: a queue of this name already exists."
+            )
+        registry.queue_info_map[self.name] = self
+
+    def enqueue(
+        self, func: "Workflow[P, R]", *args: P.args, **kwargs: P.kwargs
+    ) -> "WorkflowHandle[R]":
+        from dbos.dbos import _get_dbos_instance
+
+        dbos = _get_dbos_instance()
+        return _start_workflow(dbos, func, self.name, False, *args, **kwargs)
+
+
+def queue_thread(stop_event: threading.Event, dbos: "DBOS") -> None:
+    while not stop_event.is_set():
+        time.sleep(1)
```

Comment:
Ew

Reply:
I hypothesize that "concurrency" here is the number of workflows per VM that will be initiated per second(ish), and there is no test that establishes otherwise.

Reply:
It's correct--I updated the "one-at-a-time" test adding a sleep to confirm that additional poller ticks don't start another workflow. So we now have a test that concurrency works as advertised.

Agree this is isn't optimized--next PR will switch this over to listen/notify to make it more efficient.

Reply:
So concurrency is sorta cluster-wide